tools/azsdk-cli/docs/specs/0-v1-scenario.spec.md

```diff
@@ -0,0 +1,425 @@
+5. Execute `validate-samples` to ensure samples are valid
+6. Report generation results for each language
+
+### Update Package Metadata
+
+**Prompt:**
+```
+Update the package metadata for Health Deidentification SDKs to prepare for a preview release.
+```
+
+**Expected Agent Activity:**
+1. Execute `update-package` for all 5 languages
+2. Update version numbers for [preview release](#preview-release)
+3. Update changelogs with recent changes (only management plane)
+4. Update READMEs and language-specific metadata files
+5. Validate all updates are correctly formatted
+6. Report what was updated
+
+### Validate for PR
+
+**Prompt:**
+```
+Run all PR checks locally for the Health Deidentification SDKs before I create pull requests.
+```
+
+**Expected Agent Activity:**
+1. Execute `run-pr-checks` for all 5 languages
+2. Run build, test, lint, breaking change detection
+3. Report which checks passed and which failed
+4. Provide guidance on fixing any failures
+
+---
+
+## CLI Commands
+
+_Direct command-line interface usage for [CLI mode](#cli-mode):_
+
+### 1. Verify Setup
+
+**Command:**
+```bash
+azsdk verify-setup --languages .NET,Java,JavaScript,Python,Go
+```
+
+**Options:**
+- `--languages <list>`: Comma-separated list of languages to check (default: all)
+- `--verbose`: Show detailed output for each check
+- `--fix`: Attempt to automatically install missing dependencies
+
+**Expected Output:**
+```
+Checking environment setup for 5 languages...
+
+✓ .NET SDK 9.0.205 installed
+✓ Java JDK 11 installed
+✓ Node.js 18.x installed
+✓ Python 3.11 installed
+✗ Go 1.21 not found
+
+Environment check complete: 4/5 languages ready
+Run with --fix to install missing dependencies
+```
+
+### 2. Generate SDK
+
+**Command:**
+```bash
+azsdk generate-sdk --service healthdataaiservices --spec-path <path-to-specs> --languages .NET,Java,JavaScript,Python,Go
+```
+
+**Options:**
+- `--service <name>`: Service name to generate SDK for (required)
+- `--spec-path <path>`: Path to azure-rest-api-specs repository (required)
+- `--languages <list>`: Languages to generate (default: all)
+- `--output-path <path>`: Output directory for generated SDKs (default: current language repo)
+- `--with-tests`: Generate test files
+- `--with-samples`: Generate sample files
+- `--validate`: Build and test after generation
+
+**Expected Output:**
+```
+Generating Health Deidentification SDK for 5 languages...
+
+✓ .NET: Generated successfully
+  - Code: sdk/healthdataaiservices/Azure.Health.Deidentification
+  - Tests: 15 files
+  - Samples: 8 files
+  
+✓ Java: Generated successfully
+  - Code: sdk/healthdataaiservices/azure-health-deidentification
+  - Tests: 12 files
+  - Samples: 8 files
+
+... (similar for other languages)
+
+Generation complete: 5/5 languages successful
+```
+
+### 3. Update Package Metadata
+
+**Command:**
+```bash
+azsdk update-package --service healthdataaiservices --version 1.1.0-beta.1 --languages .NET,Java,JavaScript,Python,Go
```

Comment:
Same here for the gaps of commands and interfaces. I will include them in the spec of its own tool.

Reply:
Should we just put pseudocode as example for now and later update it to reference each individual tool’s spec when it's available?

Reply:
Yes, I added an update that calls these out as samples and they will be replaced when the spec for that specific tool is created.